src/subcommands/deployments.ts

```diff
@@ -0,0 +1,409 @@
+      // If user didn't use --id, they must have used --project, thus we already know the project-id
+      const spinner_ = wait(`Fetching deployment '${deploymentId}'...`)
+        .start();
+      const specifiedDeployment = await api.getDeployment(deploymentId);
+      if (!specifiedDeployment) {
+        spinner_.fail(
+          `The deployment '${deploymentId}' does not exist, or you don't have access to it`,
+        );
+        return Deno.exit(1);
+      }
+      spinner_.succeed(`Deployment '${deploymentId}' found`);
+      projectId = specifiedDeployment.project.id;
+    }
+    let relativePos = 0;
+    for (const prev of args.prev) {
+      relativePos -= parseInt(prev || "1");
+    }
+    for (const next of args.next) {
+      relativePos += parseInt(next || "1");
+    }
+    const relativePosString = relativePos.toLocaleString(navigator.language, {
+      signDisplay: "exceptZero",
+    });
+    const spinner = wait(
+      `Searching the deployment ${relativePosString} relative to '${deploymentId}'...`,
+    ).start();
+    const maybeBuild = await searchRelativeDeployment(
+      api.listAllDeployments(projectId),
+      deploymentId,
+      relativePos,
+    );
+    if (!maybeBuild) {
+      spinner.fail(
+        `The deployment '${deploymentId}' does not have a deployment ${relativePosString} relative to it`,
+      );
+      return Deno.exit(1);
+    }
+    build = maybeBuild;
+    spinner.succeed(
+      `The deployment ${relativePosString} relative to '${deploymentId}' is '${
+        // NULL SAFETY: the subcommand searches by deployment id, thus the build must have the deployment
+        build.deployment!.id}'`,
+    );
+  }
+
+  const spinner = wait(`Fetching deployment '${deploymentId}' details...`)
+    .start();
+
+  // Need to fetch project because the build.project does not include productionDeployment
+  [build, project, databases] = projectId
+    ? await Promise.all([
+      build ? Promise.resolve(build) : api.getDeployment(deploymentId),
+      project ? Promise.resolve(project) : api.getProject(projectId),
+      api.getProjectDatabases(projectId),
+    ])
+    : await api.getDeployment(deploymentId).then(async (build) =>
+      build
+        ? [
+          build,
+          ...await Promise.all([
+            api.getProject(build.project.id),
+            api.getProjectDatabases(build.project.id),
+          ]),
+        ]
+        : [null, null, null]
+    );
+
+  if (!build) {
+    spinner.fail(
+      `The deployment '${deploymentId}' does not exist, or you don't have access to it`,
+    );
+    return Deno.exit(1);
+  }
+  if (!project) {
+    spinner.fail(
+      `The project '${projectId}' does not exist, or you don't have access to it`,
+    );
+    return Deno.exit(1);
+  }
+  if (!databases) {
+    spinner.fail(
+      `The project '${projectId}' does not exist, or you don't have access to it`,
```

Comment:
I think this error message should somehow mention "databases", in order to make it easier to troubleshoot in case of errors.